File: back-end/app/src/main/java/com/senescyt/app/model/MotivoPermiso.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "MotivoPermiso")
public class MotivoPermiso {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "motId")
    private Long motId;

    @Column(name = "motNombre")
    private String motNombre;

    @Column(name = "motEstado")
    private int motEstado;

    @Column(name = "motDescripcionLarga",length = 2000000000)
    private String motDescripcionLarga;

    @Column(name = "motDescripcionCorta",length = 2000000000)
    private String motDescripcionCorta;

    @JsonIgnore
    @OneToMany(mappedBy = "motId")
    private List<Permisos> listaPermisos;

}

File: front-end/src/app/modelo/MotivoPermiso.ts
export class MotivoPermiso {
  motId: number;
  motNombre: string;
  motEstado: number;
  motDescripcionLarga: string;
  motDescripcionCorta: string;

  constructor(motId?: number, motNombre?: string, motEstado?: number, motDescripcionLarga?: string, motDescripcionCorta?: string) {
    (this.motId = motId || 0),
      (this.motDescripcionLarga = motDescripcionLarga || ''),
      (this.motDescripcionCorta = motDescripcionCorta || ''),
      (this.motNombre = motNombre || ''),
      (this.motEstado = motEstado || 0);
  }
}
